src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -5382,6 +5382,52 @@ public static Task Run(Action action, CancellationToken cancellationToken)
                 TaskCreationOptions.DenyChildAttach, InternalTaskOptions.None);
         }
 
+        public static Task RunAsGreenThread(Action action, CancellationToken cancellationToken)
+        {
+            Task greenThreadTask = new Task(action, cancellationToken);
+            new Thread(()=>
+            {
+                Thread.t_IsGreenThread = true;
+                // TODO? Do something like register the cancellation token as an AsyncLocal or something.
+                greenThreadTask.RunSynchronously();
+            }).Start();
+            return greenThreadTask;
+        }
+
+        public static Task RunAsGreenThread(Action action)
+        {
+            Task greenThreadTask = new Task(action);
+            new Thread(()=>
+            {
+                Thread.t_IsGreenThread = true;
+                greenThreadTask.RunSynchronously();
+            }).Start();
+            return greenThreadTask;
```

Comment:
Even though it's a placeholder, for now I'd suggest instead doing:
```suggestion
            var tcs = new TaskCompletionSource();
            new Thread(()=>
            {
                Thread.t_IsGreenThread = true;
                try
                {
                    action();
                    tcs.SetResult();
                }
                catch (Exception e)
                {
                    tcs.SetException(e);
                }
            }) { IsBackground = true }.Start();
            return tcs.Task;
```
and similarly in the other overloads.  Using Task's ctor + Start isn't a great pattern and is something we'd not have made available if we could do it over, e.g. the caller could mess things up by calling Start on the returned task.